pkg/operator/hostendpointscontroller2/host_endpoints_controller.go

```diff
@@ -117,6 +123,22 @@ func (c *HostEndpoints2Controller) syncHostEndpoints2(ctx context.Context, recor
 		})
 	}
 
+	// if the endpoint isn't present, add a check to see if we are still in a bootstrap phase (based on the etcd member).
+	// if we are, this cluster got itself in an illegal state where the endpoint is going to be created, but it won't have
+	// the bootstrap IP annotation.  Lack of this prevents successful bootstrapping and we want a better signal.  In general
+	// we want to be able to recreate this endpoint and we can do so *after* bootstrapping is complete, but not before.
+	if _, err := c.endpointsLister.Endpoints(operatorclient.TargetNamespace).Get("host-etcd-2"); errors.IsNotFound(err) {
+		members, err := c.etcdClient.MemberList()
+		if err != nil {
+			utilruntime.HandleError(err)
+		}
```

Comment:
Continuing past the point with a non-nil `err`

1. access a potentially unsafe return value from MemberList
2. fails to short-circuit when we haven't determined for sure whether bootstrap is complete

Seems safer to bail early on error from MemberList

Reply:
> Continuing past the point with a non-nil `err`
> 
> 1. access a potentially unsafe return value from MemberList
> 2. fails to short-circuit when we haven't determined for sure whether bootstrap is complete
> 
> Seems safer to bail early on error from MemberList

sure